src/hub.rs

```diff
@@ -75,25 +78,16 @@ impl<F: FnOnce() -> I, I: IntoBreadcrumbs> IntoBreadcrumbs for F {
     }
 }
 
-#[doc(hidden)]
-#[cfg(feature = "with_client_implementation")]
-pub struct PendingProcessors(
-    Vec<(
-        Option<thread::ThreadId>,
-        Box<FnOnce() -> Box<Fn(&mut Event) + Send + Sync>>,
-    )>,
-);
-
-// this is not actually sync but our uses of it are
-#[cfg(feature = "with_client_implementation")]
-unsafe impl Sync for PendingProcessors {}
 #[cfg(feature = "with_client_implementation")]
-unsafe impl Send for PendingProcessors {}
+pub enum PendingProcessor {
+    Send(Box<FnOnce() -> Box<Fn(&mut Event) + Send + Sync> + Send + Sync>),
```

Comment:
Might be nice to have `From` for this.

Reply:
Ugh. I actually should remove the pub from here I think. This should be internal only now.